code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -712,16 +712,17 @@
 /mob/living/simple_animal/proc/sentience_act() //Called when a simple animal gains sentience via gold slime potion
 	return
 
-/mob/living/simple_animal/update_sight()
+/mob/living/simple_animal/update_sight(reset_sight = FALSE)
 	if(!client)
 		return
 	if(stat == DEAD)
 		grant_death_vision()
 		return
 
-	see_invisible = initial(see_invisible)
-	see_in_dark = initial(see_in_dark)
-	sight = initial(sight)
+	if(reset_sight)

```

Comment:
But `update_sight` of simplemobs shouldn't be called every life tick - only when an effect updates the vision of the mob. Perhaps I should create an interface to set the `see_invisible` and associated vars on any mob to get this neat and consistent